scripts/user_data_pubcloud.sh

```diff
@@ -1,23 +1,104 @@
-#!/bin/bash
-
-# This file is used when creating public
-# cloud instances.
-# See playbooks/allocate_pubcloud.yml
-
-source /etc/lsb-release
-
-# Delete configuration which enables automatic upgrades
-# to prevent the instance packages being upgraded before
-# the correct apt repositories have been configured.
-# ref: RE-458 / RE-473
-if [[ "${DISTRIB_CODENAME}" == "trusty" ]]; then
-    # The 'unattended-upgrades' package cannot
-    # be purged on trusty as it is a dependency
-    # for cloud-init. So instead we just remove
-    # the configuration.
-    rm -f /etc/apt/apt.conf.d/*unattended-upgrades
+#!/bin/bash -e
```

Comment:
This user data script is applied to all cloud instances allocated via ansible. If an nodepool built image is used, it will already have this config via DIB. Therefore this script needs to be idempotent.

Reply:
Done.